SAP_Engine/Modify/OpeningType.cs

```diff
@@ -0,0 +1,170 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2023, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Xml.Serialization;
+using System.IO;
+using System.Linq;
+using System.ComponentModel;
+using BH.oM.Adapter;
+using BH.Engine.Diffing;
+using BH.oM.Environment.SAP.XML;
+using BH.oM.Environment.SAP;
+using BH.Engine.Base;
+using System.Runtime.InteropServices.ComTypes;
+using BH.oM.Environment.Elements;
+using BH.oM.Base.Attributes;
+using BH.oM.Environment.SAP.Stroma10;
+
+namespace BH.Engine.Environment.SAP
+{
+    public static partial class Modify
+    {
+        [Description("Modify opening type from a SAP report object.")]
+        [Input("sapObj", "The sap report object to modify.")]
+        [Input("type", "New type name for the windows. For each type")]
+        [Input("include", "A list of openings by name to modify.")]
+        [Input("uvalue", "Uvalue for the new type.")]
+        [Input("gvalue", "Gvalue for the new type.")]
+        [Output("sapReport", "The modified SAP Report object.")]
+        public static SAPReport ModifyOpeningTypes(this SAPReport sapObj, string type, List<string> include, double uvalue = double.NaN,  double gvalue = double.NaN)
+        {
+            if (double.IsNaN(uvalue) && double.IsNaN(gvalue))
+            {
+                return null;
+            }
+
+            if (type == null)
+            {
+                string name = string.Empty;
```

Comment:
```suggestion
                string name = "";
```

A personal preference of mine, so happy for you to ignore this one.